tools/clang/test/HLSLFileCheck/samples/SimpleHs12.hlsl

```diff
@@ -0,0 +1,76 @@
+// RUN: %dxc -E main -T hs_6_0  %s | FileCheck %s
+
+// CHECK-DAG: Patch Constant function HSPerPatchFunc should not have inout param
+// CHECK-DAG: Patch constant function's output patch input should have 3 elements, but has 5.
+
+//--------------------------------------------------------------------------------------
+// SimpleTessellation.hlsl
+//
+// Advanced Technology Group (ATG)
+// Copyright (C) Microsoft Corporation. All rights reserved.
+//--------------------------------------------------------------------------------------
+
+
+struct PSSceneIn
+{
+    float4 pos  : SV_Position;
+    float2 tex  : TEXCOORD0;
+    float3 norm : NORMAL;
+
+uint   RTIndex      : SV_RenderTargetArrayIndex;
+};
+
+
+//////////////////////////////////////////////////////////////////////////////////////////
+// Simple forwarding Tessellation shaders
+
+struct HSPerVertexData
+{
+    // This is just the original vertex verbatim. In many real life cases this would be a
+    // control point instead
+    PSSceneIn v;
+};
+
+struct HSPerPatchData
+{
+    // We at least have to specify tess factors per patch
+    // As we're tesselating triangles, there will be 4 tess factors
+    // In real life case this might contain face normal, for example
+	float	edges[ 3 ]	: SV_TessFactor;
+	float	inside		: SV_InsideTessFactor;
+};
+
+float4 HSPerPatchFunc()
+{
+    return 1.8;
+}
+
+HSPerPatchData HSPerPatchFunc( const InputPatch< PSSceneIn, 12 > points, OutputPatch<HSPerVertexData, 5> outp, inout float x)
+{
+    HSPerPatchData d;
+
+    d.edges[ 0 ] = 1;
+    d.edges[ 1 ] = 1;
+    d.edges[ 2 ] = 1;
+    d.inside = 1;
+
+    return d;
+}
+
+// hull per-control point shader
+[domain("tri")]
+[partitioning("fractional_odd")]
+[outputtopology("triangle_cw")]
+[patchconstantfunc("HSPerPatchFunc")]
+[outputcontrolpoints(3)]
+HSPerVertexData main(const uint id : SV_OutputControlPointID,
+                      const InputPatch< PSSceneIn, 12 > points )
```

Comment:
Would it not be possible to change the input patch number here, to catch both diagnostics at the same time with one run line?

Reply:
When I tried it didn't work, as the error checked in the other file makes the compile process abort earlier, and the 2 other messages don't show.

Reply:
It seems a bit odd that these are caught at different points now so I'd probably want to move all 3 checks to Sema, or at least track the remaining 2 in a follow-up bug, but I'll leave it to MS folks to decide.

Reply:
Created 2 issues to track those. One is handled by the SPIR-V backend, but weirdly, the other also causes us to accept bad code.